### PR TITLE
Add migration to populate voucher_code in BaseDiscount

### PR DESCRIPTION
**Pull request by michal-macioszczyk**
I want to merge this change because I want to add migration to populate voucher_code from existing voucher field in BaseDiscount class so we need to update all related models like `OrderDiscount`, `OrderLineDiscount`, `CheckoutLineDiscount`
https://github.com/saleor/saleor/issues/14341

# Impact

- [ ] New migrations
- [ ] New/Updated API fields or mutations
- [ ] Deprecated API fields or mutations
- [ ] Removed API types, fields, or mutations

# Docs

- [ ] Link to documentation:

# Pull Request Checklist

- [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
- [ ] Database queries are optimized and the number of queries is constant
- [ ] Database migrations are either absent or optimized for zero downtime
- [ ] The changes are covered by test cases

### CODE REVIEW

File: saleor/discount/migrations/0067_orderdiscount_voucher_to_voucher_code_migration.py
Comment (IKarbowiak) [inline saleor/discount/migrations/0067_orderdiscount_voucher_to_voucher_code_migration.py:1]: No need to write the `migration` in the migration file name

File: saleor/discount/migrations/0067_orderdiscount_voucher_to_voucher_code_migration.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+# OrderLineDiscount Migration took 27.98 seconds.
+# OrderLineDiscount Memory usage increased by 47.99 MiB.
+# CheckoutLineDiscount Migration took 28.12 seconds.
+# CheckoutLineDiscount Memory usage increased by 35.52 MiB.
+
+
+BATCH_SIZE = 1000
+
+
+def set_voucher_to_voucher_code_in_model(ModelName, apps, schema_editor):
```
Comment (IKarbowiak) [inline saleor/discount/migrations/0067_orderdiscount_voucher_to_voucher_code_migration.py:25]: Also the part `set_voucher_to_voucher_code` in the function name is not adequate. Consider maybe something like `set_voucher_code_in_model` or `assign_voucher_code_in_model`

File: saleor/discount/migrations/0065_orderdiscount_voucher_code.py
Comment (IKarbowiak) [inline saleor/discount/migrations/0065_orderdiscount_voucher_code.py:1]: Name of the mutation file should be changed, as it's extended with `CheckoutLineDiscount` and `OrderLineDiscount` changes

File: saleor/discount/migrations/0066_orderdiscount_voucher_code_add_index.py
Comment (IKarbowiak) [inline saleor/discount/migrations/0066_orderdiscount_voucher_code_add_index.py:1]: Same here 👮‍♀️

File: saleor/discount/migrations/0067_orderdiscount_voucher_to_voucher_code_migration.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+# For batch size 1000 with 100_000 per model OrderDiscount/Voucher/VoucherCode objects
+# OrderDiscount Migration took 28.68 seconds.
+# OrderDiscount Memory usage increased by 163.00 MiB.
+# OrderLineDiscount Migration took 27.98 seconds.
+# OrderLineDiscount Memory usage increased by 47.99 MiB.
+# CheckoutLineDiscount Migration took 28.12 seconds.
+# CheckoutLineDiscount Memory usage increased by 35.52 MiB.
+
+
+BATCH_SIZE = 1000
```
Comment (zedzior) [inline saleor/discount/migrations/0067_orderdiscount_voucher_to_voucher_code_migration.py:22]: I am not sure if this is a proper size o the batch. According to docs: https://docs.saleor.io/docs/3.x/developer/community/zero-downtime-migrations#solution, single batch shouldn't take more then one second.

Comment (michal-macioszczyk) [reply saleor/discount/migrations/0067_orderdiscount_voucher_to_voucher_code_migration.py:22]: zero downtown migration will be done in different ticket, those times are for whole migration.

File: saleor/discount/migrations/0067_orderdiscount_voucher_to_voucher_code_migration.py
Comment (michal-macioszczyk) [reply saleor/discount/migrations/0067_orderdiscount_voucher_to_voucher_code_migration.py:1]: done

File: saleor/discount/migrations/0067_orderdiscount_voucher_to_voucher_code_migration.py (hunk 1 above)
Comment (michal-macioszczyk) [reply saleor/discount/migrations/0067_orderdiscount_voucher_to_voucher_code_migration.py]: changed to set_voucher_code_in_model

File: saleor/discount/migrations/0066_orderdiscount_voucher_code_add_index.py
Comment (michal-macioszczyk) [reply saleor/discount/migrations/0066_orderdiscount_voucher_code_add_index.py:1]: done

File: saleor/discount/migrations/0065_orderdiscount_voucher_code.py
Comment (michal-macioszczyk) [reply saleor/discount/migrations/0065_orderdiscount_voucher_code.py:1]: done

**Review by IKarbowiak — APPROVED** [review]

**Review by SzymJ — APPROVED** [review]

**Pull request merged**